src/__experimental__/components/numeral-date/numeral-date.component.js

```diff
@@ -0,0 +1,140 @@
+import React, { useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import invariant from 'invariant';
+import Events from '../../../utils/helpers/events';
+import StyledNumeralDate, { StyledDateField } from './numeral-date.style';
+import Textbox from '../textbox';
+
+const NumeralDate = ({
+  dateFormat,
+  defaultValue,
+  errorMessage,
+  errorPresent,
+  id,
+  name,
+  onBlur,
+  onChange,
+  value
+}) => {
+  const isControlled = value !== undefined;
+  const initialValue = isControlled ? value : defaultValue;
+  invariant(
+    initialValue !== undefined,
+    'This component has no initial value'
+  );
+
+  let inputRef = useRef();
+  const [isActive, setIsActive] = useState(inputRef.current === document.activeElement);
+  const [dateValue, setDateValue] = useState({
+    ...initialValue
+  });
+
+  const isValidKeyPress = (ev) => {
+    if (Events.isNumberKey(ev) || Events.isDeletingKey(ev) || Events.isTabKey(ev)) {
+      return true;
+    }
+    return false;
+  };
+
+  const onKeyPress = (ev) => {
+    if (!isValidKeyPress(ev)) {
+      ev.preventDefault();
+    }
```

Comment:
```suggestion
    const isValidKey = Events.isNumberKey(ev) || Events.isDeletingKey(ev) || Events.isTabKey(ev);
    
    if(isValidKey) {
      ev.preventDefault();
    }
```

Reply:
Yeah I get your recommendation however it would have to be:

```
if (!isValidKey) {
ev.preventDefault();
}
```